services/QuillLMS/spec/controllers/teachers/progress_reports/diagnostic_reports_controller_spec.rb

```diff
@@ -403,4 +403,56 @@
     end
 
   end
+
+  describe '#student_ids_for_previously_assigned_activity_pack' do
+    let!(:activity_pack) { create(:unit_template) }
+    let!(:student1) { create(:student, name: 'Alphabetical A')}
+    let!(:students_classroom1) { create(:students_classrooms, classroom: classroom, student: student1)}
+    let!(:student2) { create(:student, name: 'Alphabetical B')}
+    let!(:students_classroom2) { create(:students_classrooms, classroom: classroom, student: student2)}
+    let!(:unit_based_on_activity_pack1) { create(:unit, user: teacher, unit_template: activity_pack, name: 'First Unit') }
+    let!(:classroom_unit1) { create(:classroom_unit, classroom: classroom, unit: unit_based_on_activity_pack1, assigned_student_ids: [student1.id] ) }
+    let!(:unit_based_on_activity_pack2) { create(:unit, user: teacher, unit_template: activity_pack, name: 'Second Unit') }
+    let!(:classroom_unit2) { create(:classroom_unit, classroom: classroom, unit: unit_based_on_activity_pack2, assigned_student_ids: [student2.id] ) }
+
+    it 'should return all the student ids for the assigned pack' do
+      get :student_ids_for_previously_assigned_activity_pack, params: { classroom_id: classroom.id, activity_pack_id: activity_pack.id }
+
+      expect(response).to be_successful
+      json = JSON.parse(response.body)
+      expect(json['student_ids']).to eq [student1.id, student2.id]
+    end
+  end
+
+  describe '#assign_post_test' do
+    let!(:activity_pack) { create(:unit_template) }
+    let!(:student1) { create(:student, name: 'Alphabetical A')}
+    let!(:students_classroom1) { create(:students_classrooms, classroom: classroom, student: student1)}
+    let!(:student2) { create(:student, name: 'Alphabetical B')}
+    let!(:students_classroom2) { create(:students_classrooms, classroom: classroom, student: student2)}
+
+    it 'creates a new pack and assigns it if there is no existing pack' do
+      get :assign_post_test, params: { classroom_id: classroom.id, unit_template_id: activity_pack.id, student_ids: [student1.id] }
```

Comment:
I think that this might be worth doing an explicit expectation about a new record being created.  We can do that by wrapping this in an `expect ... to change` block:
```
expect do
  get :assign_post_test, params: { classroom_id: classroom.id, unit_template_id: activity_pack.id, student_ids: [student1.id] }
end.to change(ClassroomUnit, :count).by(1)
```